src/main/java/bio/terra/service/dataset/flight/create/CreateDatasetGetOrCreateStorageAccountStep.java

```diff
@@ -0,0 +1,61 @@
+package bio.terra.service.dataset.flight.create;
+
+import bio.terra.model.BillingProfileModel;
+import bio.terra.model.CloudPlatform;
+import bio.terra.model.DatasetRequestModel;
+import bio.terra.service.dataset.DatasetJsonConversion;
+import bio.terra.service.dataset.flight.DatasetWorkingMapKeys;
+import bio.terra.service.profile.flight.ProfileMapKeys;
+import bio.terra.service.resourcemanagement.AzureDataLocationSelector;
+import bio.terra.service.resourcemanagement.ResourceService;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateDatasetGetOrCreateStorageAccountStep implements Step {
```

Comment:
Does this class name mean that it can only be used inside a CreateDataset flight? If we had another flight that needs to get or create a storage account, would its step be different from this one?

Reply:
I still have questions about how we name steps.. maybe most steps are very flight specific? I guess given how inputs and outputs are handled (fixed keys in a hash map) they would have to be.